File: part-2-simple-ut-particle/python/particle.py
class Particle:
    def __init__(self, pos_x, vx):
        self.mass = 1.0
        self.pos_x = pos_x
        self.vx = vx
    
    def get_x(self):
        return self.pos_x
    
    def get_vx(self):
        return self.vx

    def move(self, dt):
        self.pos_x += dt * self.vx

    def set_vx(self, vx):
        self.vx = vx

    def get_mass(self):
        return self.mass

File: part-2-simple-ut-particle/cpp/particle.cpp
/*********************************************************/
#include "particle.hpp"

/*********************************************************/
using namespace experiment;

/*********************************************************/
Particle::Particle(double pos_x, double vx)
         :pos_x(pos_x)
         ,vx(vx)
         ,mass(1.0)
{
}

/*********************************************************/
double Particle::get_x(void) const
{
	return this->pos_x;
}

/*********************************************************/
double Particle::get_vx(void) const
{
	return this->vx;
}

/*********************************************************/
double Particle::get_mass(void) const
{
	return this->mass;
}

/*********************************************************/
void Particle::set_vx(double vx)
{
	this->vx = vx;
}

/*********************************************************/
void Particle::move(double dt)
{
	this->pos_x += this->vx * dt;
}
